### ISSUE TITLE
Does this work with remote Docker hosts?

### ISSUE THREAD

**Issue by meichthys** [ualex73/monitor_docker#6]
From what I've read, this integration is meant to report the stats from containers running on the same machine as home-assistant.

Is it possible to use this integration with a remote docker host?
Something like:
```
# Example configuration.yaml entry
monitor_docker:
  - host: 192.168.0.11
    name: Docker
    containers:
      - appdaemon
      - db-dsmr
    rename:
      appdaemon: AppDaemon
      db-dsmr: "Database DSMR-Reader"
    monitored_conditions:
      - version
      - containers_running
      - containers_total
      - status
      - memory
```

**Comment by ualex73** [comment]
Yes, it should work (only I have tested it myself yet). The underlying aiodocker library supports it.

You can modify your configuration as follows with the `url`:
```
monitor_docker:
  - url: http://192.168.0.11:2376/
    name: Docker
    containers:
...
```

**Comment by meichthys** [comment]
@ualex73 Great news!
I tried adding the 'url' parameter like  you suggested:
```
# configuration.yaml entry
monitor_docker:
  - url: tcp://10.0.7.12:2376/
    name: Docker
    containers:
      - container_name
```
but i get the following:
```aiodocker.exceptions.DockerError: DockerError(900, "Cannot connect to Docker Engine via tcp://10.0.7.12:2376/ [Connect call failed ('10.0.7.12', 2376)]")```

**Comment by ualex73** [comment]
I will try to build a docker server with a remote tcp connection this tomorrow/weekend and test it.

Does the following work?
```docker -H tcp://10.0.7.12:2376 ps```

If that doesn't work, then you need to investigate that. You sure your docker is listening on port 2376? it could be something else too.

**Comment by meichthys** [comment]
@ualex73 Indeed, the `docker -H tcp://10.0.7.12:2376 ps` command is not working on my machine. I'll need to look into that if i get a chance and report back.

**Comment by IAmTheDude** [comment]
I ran into this issue using Sanderhuisman version of this, I found a workaround using another docker container to espose the tcp ports (both securely and unsecurely) which I am still using.

https://github.com/Sanderhuisman/docker_monitor/issues/19#issuecomment-525962793

I used it on my Proxmox version of HAOS (Whiskey script) to get the conatiner running using Portainer as for the life of me could not get it to access the underlying docker on the install without some messing with the Supervisor and homeassistant containers.

I don't know if this is still the best way to do this but it works and saves messing with editing files etc and has worked on every docker instance I have with no problem.

**Comment by ualex73** [comment]
@meichthys I tested it on my setup with a TCP connection, and it works 👍 Ofcourse, the "docker -H ... ps" need to work first.

**Comment by ualex73** [comment]
@IAmTheDude Thanks for the comments, I will investigate TLS (SSL), then I can implement that open too.

**Comment by ualex73** [comment]
@IAmTheDude I don't have a test Docker with TLS, but if you change the url to:
```https://ip:2376```

Then it should connect via TLS, according to the underlying code of aiodocker.

**Comment by elRadix** [comment]
how to get it working tls? 

```
[custom_components.monitor_docker.helpers]
Can not connect to Docker API (Missing protocol scheme in docker_host.)


monitor_docker 
  - name: RemoteDocker
    url: https://docker.lan:2376
```

I have setup the certs according to this guide on the docker vm (proxmox) 
https://docs.docker.com/engine/security/https/

imported the certs on the client machine below and execute the command :
![IMG_20200621_125921_147](https://user-images.githubusercontent.com/6258894/85222954-2ccd5100-b3bf-11ea-82c5-be8dab1ed433.jpg)

**Comment by ualex73** [comment]
Looked into the TLS code again, and https is set afterwards (oeps). I don't have TLS set, so it is difficult for me to test it.

I *think* the following should work:
(1) set *url* to tcp://docker.lan:2376 (http:// is also allowed)

(2) set the following environment variables in Docker/your environment:
DOCKER_TLS_VERIFY=1
DOCKER_CERT_PATH=<path to your certificates> (should be "/etc/docker/certs/" in your example)

**Comment by meichthys** [comment]
I’ll take another look at this this afternoon, but I’ll still working on getting a response when running the ‘docker tls-verify’ command.

**Comment by ualex73** [comment]
Strange, if you set the environment values correctly (that is what I tested), it tries to connect with HTTPS/TLS. Not sure where this error comes from. I am not sure what wrong here, it almost looks like the Home Assistant instance doesn't have the environment variables set properly. I will add this type of information to the debug logging, makes troubleshooting easier.

**Comment by ualex73** [comment]
Can you replace the "helpers.py" with the one in the repository? You can do this like:
```
cd ~/custom_components/monitor_docker
rm helpers.py
wget https://raw.githubusercontent.com/ualex73/monitor_docker/master/custom_components/monitor_docker/helpers.py
```

Also add the following your logging section of HA:
```
logger:
  logs:
    custom_components.monitor_docker: debug
```
And then restart your HA.

Then you need to see something like:
```
2020-06-21 18:16:24 DEBUG (Thread-2) [custom_components.monitor_docker.helpers] Docker URL is 'tcp://192.168.1.1:2376'
2020-06-21 18:16:24 DEBUG (Thread-2) [custom_components.monitor_docker.helpers] Docker environment set for 'DOCKER_TLS_VERIFY=1'
2020-06-21 18:16:24 DEBUG (Thread-2) [custom_components.monitor_docker.helpers] Docker environment set for 'DOCKER_CERT_PATH=/var/tmp/cert'
```

If you don't see those 2 last lines, there is an issue with your environment variables.

**Comment by elRadix** [comment]
Really strange my environment variables are correct as per screenshot below. I am running home assistant as venv setup.

![Screenshot_20200621-194453_JuiceSSH](https://user-images.githubusercontent.com/6258894/85231666-8bafbc00-b3f9-11ea-8042-1e7822c574f8.jpg)

It still thinks the environment are not set
![IMG_20200621_195036_833](https://user-images.githubusercontent.com/6258894/85231672-91a59d00-b3f9-11ea-8e29-d5f4d61d0cdc.jpg)

**Comment by ualex73** [comment]
This confirms the environment variables aren't passed to HA. Let me see if I can make a workaround for your problem in the integration itself. I added some code to the master branch (const.py/__init__.py/helpers.py you need to download) to workaround your environment issues.

I added a 'certpath' parameter, this will explicitly set the docker environment variables for you.

```
monitor_docker:
  - name: Docker
    url: tcp://192.168.1.1:2376
    certpath: /etc/certpath
    containers:
      - appdaemon
```

**Comment by elRadix** [comment]
@ualex73 just tested, it now works
thank you very much for your patience and follow up.

**Comment by ualex73** [comment]
@elRadix Thanks for the confirmation and testing it (I couldn't test the TLS option myself). This "certpath" option will be also included in the next official release in HACS (currently it is only in the main repository).

Yes, i agree the TLS is only option to go when using TCP (there doesn't seem to be any other authentication option?).

**Comment by ualex73** [comment]
I believe all is fixed in the latest 1.0/1.1 releases. Please open a new issue if the problem persist, with 'fresh' logging/information.
